.buildkite/diff

```diff
@@ -8,13 +8,18 @@
 
 COMMIT=$1
 
-HEAD_BRANCH=$(git remote show origin | awk '/HEAD branch/ {print $NF}')
-BRANCH_POINT_COMMIT=$(git merge-base "$HEAD_BRANCH" "$COMMIT")
-
-if [ "$BUILDKITE_BRANCH" == "$HEAD_BRANCH" ]; then
-	echo "diff between $COMMIT and HEAD~1"
-        git diff --raw HEAD~1 | awk '{print $6; if($7) {print $7}}'
+if [ -n $BUILDKITE_PULL_REQUEST_BASE_BRANCH ]; then
+  # git repository here is a shallow clone. Because of it, we need to fetch
+  # the base branch before doing the diff.
+  # We fetch last 100 commits of the branch, probably smaller will be fine
+  # but for now this costs nothing.
+  # We might reconsider this code later if problems happen.
+  git fetch origin $BUILDKITE_PULL_REQUEST_BASE_BRANCH --depth=100
```

Comment:
`--depth=100` might not work if the testing branch has more than 100 commits.

Reply:
I actually removed the whole fetch and it works well: https://buildkite.com/elastic/connectors-python/builds/9960#018b2967-e5c3-4179-a012-ccb5ccd4cc18

```
.buildkite/diff 2c13b21d24e616c4ce698de18c93cc4a6f197cd0
--
  | Checking against a base branch: 8.10 with merge base at c4892ff5c9b20871ececde956145445eddac0afa
  | .buildkite/diff
  | .buildkite/pipeline.yml
  | tests/sources/fixtures/mysql/fixture.py


```